src/platform_impl/macos/observer.rs

```diff
@@ -169,31 +179,50 @@ impl Default for EventLoopWaker {
                 ptr::null_mut(),
             );
             CFRunLoopAddTimer(CFRunLoopGetMain(), timer, kCFRunLoopCommonModes);
-            EventLoopWaker { timer }
+            EventLoopWaker {
+                timer,
+                start_instant: Instant::now() - Duration::from_secs(1000),
```

Comment:
What's the point of such a giant sub? At the very least it must be documented.

Reply:
It's an arbitrary instant that we know is in the past.

`start_instant` itself is documented with:
```
    /// An arbitrary instant in the past, that will trigger an immediate wake
    /// We save this as the `next_fire_date` for consistency so we can
    /// easily check if the next_fire_date needs updating.
```

Reply:
I mean doc comment means during use, but `start_instant` during runtime is something different.

Reply:
Added a small comment where it's initialized that just notes that it's `// An arbitrary instant in the past`